contracts/performance/src/storage.rs

```diff
@@ -482,10 +542,12 @@ impl PerformanceResultsStorage {
         epoch_id: EpochId,
         data: &NodePerformance,
     ) -> Result<(), NymPerformanceContractError> {
+        self.assert_measurement_defined(storage, data.measurement_kind.clone())?;
```

Comment:
this function should only really be concerned about writing data to storage and not checking whether it's actually allowed to insert it (i.e. ensuring kind exists). imo that should be done by the caller